core/src/main/java/com/amazon/opendistroforelasticsearch/sql/analysis/ExpressionAnalyzer.java

```diff
@@ -33,21 +34,25 @@
 import com.amazon.opendistroforelasticsearch.sql.expression.function.BuiltinFunctionName;
 import com.amazon.opendistroforelasticsearch.sql.expression.function.BuiltinFunctionRepository;
 import com.amazon.opendistroforelasticsearch.sql.expression.function.FunctionName;
-import java.util.List;
-import java.util.stream.Collectors;
-import lombok.RequiredArgsConstructor;
-
+import java.util.Arrays;
 import java.util.Collections;
+import java.util.List;
 import java.util.Optional;
+import java.util.stream.Collectors;
 
 /**
  * Analyze the {@link UnresolvedExpression} in the {@link AnalysisContext} to construct the {@link
  * Expression}
  */
-@RequiredArgsConstructor
 public class ExpressionAnalyzer extends AbstractNodeVisitor<Expression, AnalysisContext> {
-  private final DSL dsl;
   private final BuiltinFunctionRepository repository;
+  private final DSL dsl;
+
+  public ExpressionAnalyzer(
+      BuiltinFunctionRepository repository) {
+    this.repository = repository;
+    this.dsl = new DSL(repository);
```

Comment:
Is the `DSL` bean useless in `ExpressionConfig`?

Reply:
no, it still been used.